tests/test_site_apportionment/test_site_apportionment.py

```diff
@@ -634,3 +643,87 @@ def test_sort_rows_missing_cols(self, caplog):
 
         with pytest.raises(KeyError):
             sort_rows_order_cols(input_df, cols_in_order)
+
+
+@pytest.fixture(scope="function")
+def sites_df_input():
+    """
+    Input df for create_sites_df tests.
+    
+    This input data consists of:
+    1. Columns that will be dropped
+    2. NaN's in used columns (601, 602)
+    3. Combinations of Postcodes/Periods/References to create multiple groups
+    """
+    input_cols = [
+        "reference",
+        "instance",
+        "formtype",
+        "601",
+        "602",
+        "601_count",
+        "status",
+        "imp_marker",
+        "postcodes_harmonised",
+        "period"
+    ]
+    input_data = [
+        [1, 0, "0006", "RH12 1XL", 100.0, np.nan, "Clear", "R", "RH12 1XL", "202101"],
+        [1, 1, "0006", "RH12 1XL", 125.0, np.nan, "Clear", "R", "RH12 1XL", "202101"],
+        [1, 2, "0006", "RH12 1XL", np.nan, np.nan, "Clear", "R", "RH12 1XL", "202101"], # Nan 602 - Ensure conv to 0
+        [1, 3, "0006", "RH12 1XZ", 100.0, np.nan, "Clear", "R", "RH12 1XZ", "202101"], # different postcode
+        [2, 0, "0001", "NP44 2NZ", np.nan, 2.0, "Clear", "R", "NP44 2NZ", "202102"],
+        [2, 1, "0001", "NP44 2NZ", 50.0, 2.0, "Clear", "R", "NP44 2NZ", "202102"],
+        [3, 0, "0001", np.nan, np.nan, 1.0, "Check needed", "TMI", "NP30 7ZZ", "202102"], # NaN 601 - Ensure dropped
+    ]
```

Comment:
For illustration of what the pipeline is like at this point, please don't include formtype or instance 0